sdk/src/azure/core/az_log.c

```diff
@@ -29,6 +29,27 @@ void az_log_set_callback(az_log_message_fn az_log_message_callback)
   _az_log_message_callback = az_log_message_callback;
 }
 
+// Verifies that  the classification that the user provided is one of the valid possibilties
+// and guards against looping past the end of the classification array.
+// Make sure to update the switch statement whenever new classifications are added.
+static bool _az_is_valid_log_classification(az_log_classification const* classification)
```

Comment:
I think it is better to be AZ_INLINE to eliminate the invocation cost.

Reply:
Making it a precondition for now, so leaving it as is.

Reply:
Precondition should not interfere with inline. I see no reason to not make it such.